plugins/BEdita/Core/src/Model/Behavior/CustomPropertiesBehavior.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Model\Behavior;
+
+use Cake\Datasource\EntityInterface;
+use Cake\Event\Event;
+use Cake\ORM\Behavior;
+use Cake\ORM\Query;
+use Cake\ORM\TableRegistry;
+
+/**
+ * CustomProperties behavior
+ *
+ * @since 4.0.0
+ */
+class CustomPropertiesBehavior extends Behavior
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected $_defaultConfig = [
+        'field' => 'custom_props',
+    ];
+
+    /**
+     * The custom properties available.
+     * It is an array with properties name as key and Property entity as value
+     *
+     * @var array
+     */
+    protected $available = null;
+
+    /**
+     * Get available properties for object type
+     *
+     * @return array
+     */
+    public function getAvailable()
+    {
+        if ($this->available !== null) {
+            return $this->available;
+        }
+
+        try {
+            $objectType = TableRegistry::get('ObjectTypes')->get($this->getTable()->getAlias());
+            // @todo add cache for properties
+            $props = TableRegistry::get('Properties')->find()
+                ->where([
+                    'object_type_id' => $objectType->id,
+                    'enabled' => 1,
+                ])
+                ->all();
+        } catch (\Exception $ex) {
+            $props = false;
+        }
+
+        if (!$props) {
+            return $this->available = [];
```

Comment:
As far as I can tell, this "if" construct is redundant, as its inner lines of codes can only be reached when the "try/catch" construct above fell into the "catch" branch.

Long story short: I'd just put `return $this->available = [];` in the "catch" and remove this "if". 🙃  It'd also lower cyclomatic complexity as a bonus.

Reply:
done